### PR TITLE
Add output for `prebuilt_android_aar`

### PR DESCRIPTION
**Pull request by rowillia**
--show_output currently returns null for AARs.  This fixes it.

### CODE REVIEW

File: src/com/facebook/buck/android/UnzipAar.java [hunk 1]
```diff
@@ -203,7 +213,7 @@ public BuildOutput initializeFromDisk(OnDiskBuildInfo onDiskBuildInfo) throws IO
   @Override
   @Nullable
```
Comment (sdwilsh) [inline src/com/facebook/buck/android/UnzipAar.java]: This is no longer `Nullable`

File: test/com/facebook/buck/android/AndroidPrebuiltAarIntegrationTest.java [hunk 2]
```diff
@@ -73,6 +74,16 @@ public void testAndroidPrebuiltAarInDepsIsExported() throws IOException {
   }
 
   @Test
+  public void testAndroidPrebuiltAarHasOutput() throws IOException {
+    ProjectWorkspace.ProcessResult targetsResult = workspace.runBuckCommand(
+        "targets",
+        "--show_output",
+        "//:aar");
+    targetsResult.assertSuccess();
+    assertEquals("//:aar buck-out/gen/aar#aar_unzip.aar", targetsResult.getStdout().trim());
```
Comment (sdwilsh) [inline test/com/facebook/buck/android/AndroidPrebuiltAarIntegrationTest.java]: You hate Windows, don't you? :P

File: src/com/facebook/buck/android/AndroidPrebuiltAar.java [hunk 3]
```diff
@@ -119,6 +121,12 @@ public Path getBinaryJar() {
     return prebuiltJar.getPathToOutput();
   }
 
+  @Nullable
```
Comment (sdwilsh) [inline src/com/facebook/buck/android/AndroidPrebuiltAar.java]: This shouldn't be `Nullable`

**Comment by sdwilsh** [comment]
I'll fix those comments up before merging.

File: test/com/facebook/buck/android/AndroidPrebuiltAarIntegrationTest.java (hunk 2 above)
Comment (rowillia) [inline test/com/facebook/buck/android/AndroidPrebuiltAarIntegrationTest.java]: With a burning passion.

File: src/com/facebook/buck/android/UnzipAar.java [hunk 4]
```diff
@@ -203,7 +213,7 @@ public BuildOutput initializeFromDisk(OnDiskBuildInfo onDiskBuildInfo) throws IO
...
   public Path getPathToOutput() {
-    return null;
+    return copiedAar;
```
Comment (andrewjcg) [inline src/com/facebook/buck/android/UnzipAar.java]: It seems odd that we're listing essentially a copy of the input file as the output.  Shouldn't the output of `UnzipAar` be `unpackDirectory`?

File: src/com/facebook/buck/android/AndroidPrebuiltAar.java [hunk 5]
```diff
@@ -119,6 +121,12 @@ public Path getBinaryJar() {
...
+  @Override
+  public Path getPathToOutput() {
+    return unzipAar.getPathToOutput();
```
Comment (andrewjcg) [inline src/com/facebook/buck/android/AndroidPrebuiltAar.java]: It feels a bit round-about to use the `UnzipAar` rule to thread through the original AAR (but it's likely I'm missing something).  Can we just have the constructor of this rule take the orignal AAR `SourcePath` and list that as the output without touching `UnzipAar` (and also propagate the appropriate runtime dep on any rule that generates the original AAR)?

Comment (rowillia) [inline src/com/facebook/buck/android/AndroidPrebuiltAar.java]: The reason I went down this path is `JavaBinary` followed a similar pattern of first copying the input jar to `buck-out` before doing anything with it.

File: src/com/facebook/buck/android/UnzipAar.java (hunk 4 above)
Comment (rowillia) [inline src/com/facebook/buck/android/UnzipAar.java]: Yeah, it does feel a little janky.  I guess we could graph enhance another step in there to copy the input first and then pass that to the prebuilt jar and aar?

Comment (sdwilsh) [inline src/com/facebook/buck/android/UnzipAar.java]: That would probably be best.

**Comment by Coneko** [comment]
Back to @andrewjcg for @rowillia's comment.

**Comment by facebook-github-bot-6** [comment]
Thank you for your pull request.  As you may know, we require contributors to sign our Contributor License Agreement, and we don't seem to have you on file and listed as active anymore.  In order for us to review and merge your code, please email cla@fb.com with your details so we can update your status.

**Comment by styurin** [comment]
@rowillia this PR is pretty old. Please, rebase and resubmit it if you want to merge these changes in.

**Pull request closed without merging**